CMakeLists.txt

```diff
@@ -435,114 +480,84 @@ function(is_sdk_requested name result_var_name)
   endif()
 endfunction()
 
-# FIXME: separate the notions of SDKs used for compiler tools and target
-# binaries.
-if("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux")
-  set(CMAKE_EXECUTABLE_FORMAT "ELF")
+# Required from this point:
+# - SWIFT_HOST_VARIANT_SDK  - OS of system which will host the compiler
+# - SWIFT_HOST_VARIANT_ARCH - arch of system which will host the compiler
+# 
+# FIXME: cross-compile standard libarary targets are specified by "SDK",
+# which locks each one to the hardcoded architectures defined here (e.g. armv7 for ANDROID)
```

Comment:
Also, to be truly flexible we would need to pass along the sysroot/SDK path for every SDK/arch combination. Darwin targets currently cheat and use `xcrun`, Android passes it in as a specific parameter. That's not flexible - what if I wanted to cross-compile the stdlib for multiple linux targets? Each would have a different sysroot, and we'd need to tell it, for example:

`-DSWIFT_SDKs="LINUX-x86_64;LINUX-armv6;LINUX-armv7;ANDROID-armv7"
-DSWIFT_SDK_LINUX_x86_64_SYSROOT="/"
-DSWIFT_SDK_LINUX_armv6_SYSROOT="/sysroots/debian.jessie.armv6"
-DSWIFT_SDK_LINUX_armv7_SYSROOT="/sysroots/debian.jessie.armv7"
-DSWIFT_SDK_ANDROID_armv7_SYSROOT="/sysroots/android-ndk-armv7"`